yellowbrick/features/projection.py

```diff
@@ -286,6 +286,8 @@ def finalize(self):
             if self.colorbar:
                 if self.projection == 3:
                     sm = plt.cm.ScalarMappable(cmap=self._colors, norm=self._norm)
+                    # Avoid MPL TypeError: "You must first set_array for mappable"
+                    sm.set_array([])
```

Comment:
I've confirmed that the scalar values are correct for the concrete dataset, e.g. that the range of the labels in the color bar matches with this call to `set_array` and without. I've only been able to get the `TypeError` once, so it's not reproducible for me generally. 

I think we should move forward with this PR but make an issue to dive into this deeper. Creating colorbars without data seems to be a debated feature in matplotlib:

https://github.com/matplotlib/matplotlib/issues/3644

It might be as simple that we need to set the range in the `ScalarMappable` and keep the minimum and maximum of the data around so that we don't have to keep the data. It also might be as simple as us being able to use `self.fig` instead of `plt`. 

Either way - this PR is to important to block as it is blocking all of our other PRs.

Reply:
Thanks again @rebeccabilbro for all of your hard work on this and taking care of this so quickly and @bbengfort for adding your review.  I will go ahead and merge this in.